code/modules/integrated_electronics/core/integrated_circuit.dm

```diff
@@ -0,0 +1,399 @@
+		var/obj/item/integrated_circuit/examined
+		if(href_list["examined"])
+			examined = href_list["examined"]
+		else
+			examined = src
+		examined.interact(usr)
+		update = 0
+
+	if(href_list["bottom"])
+		var/obj/item/integrated_circuit/circuit = locate(href_list["bottom"]) in src.assembly.contents
+		var/assy = circuit.assembly
+		if(!circuit)
+			return
+		circuit.loc = null
+		circuit.loc = assy
+		. = 1
+		update_to_assembly = 1
+
+	if(href_list["scan"])
+		if(istype(held_item, /obj/item/device/integrated_electronics/debugger))
+			var/obj/item/device/integrated_electronics/debugger/D = held_item
+			if(D.accepting_refs)
+				D.afterattack(src, usr, TRUE)
+			else
+				to_chat(usr, "<span class='warning'>The Debugger's 'ref scanner' needs to be on.</span>")
+		else
+			to_chat(usr, "<span class='warning'>You need a multitool/debugger set to 'ref' mode to do that.</span>")
+
+	if(href_list["return"])
+		if(A)
+			update_to_assembly = 1
+			usr << browse(null, "window=circuit-\ref[src];border=1;can_resize=1;can_close=1;can_minimize=1")
+		else
+			to_chat(usr, "<span class='warning'>This circuit is not in an assembly!</span>")
+
+
+	if(href_list["remove"])
+		if(!A)
+			to_chat(usr, "<span class='warning'>This circuit is not in an assembly!</span>")
+			return
+		if(!removable)
+			to_chat(usr, "<span class='warning'>\The [src] seems to be permanently attached to the case.</span>")
+			return
+		var/obj/item/device/electronic_assembly/ea = loc
+		disconnect_all()
+		var/turf/T = get_turf(src)
+		forceMove(T)
+		assembly = null
+		playsound(T, 'sound/items/Crowbar.ogg', 50, 1)
+		to_chat(usr, "<span class='notice'>You pop \the [src] out of the case, and slide it out.</span>")
+
+		if(istype(ea))
+			ea.interact(usr)
+		update = 0
+		return
+
+	if(update)
+		if(A && istype(A) && update_to_assembly)
+			A.interact(usr)
+		else
+			interact(usr) // To refresh the UI.
+
+
+
+/obj/item/integrated_circuit/proc/push_data()
+	for(var/datum/integrated_io/O in outputs)
+		O.push_data()
+
+/obj/item/integrated_circuit/proc/pull_data()
+	for(var/datum/integrated_io/I in inputs)
+		I.push_data()
+
+/obj/item/integrated_circuit/proc/draw_idle_power()
+	if(assembly)
+		return assembly.draw_power(power_draw_idle)
+
+// Override this for special behaviour when there's no power left.
+/obj/item/integrated_circuit/proc/power_fail()
+	return
+
+// Returns true if there's enough power to work().
+/obj/item/integrated_circuit/proc/check_power()
+	if(!assembly)
+		return FALSE // Not in an assembly, therefore no power.
+	if(assembly.draw_power(power_draw_per_use))
+		return TRUE // Battery has enough.
+	return FALSE // Not enough power.
+
+/obj/item/integrated_circuit/proc/check_then_do_work(var/ignore_power = FALSE)
+	if(world.time < next_use) 	// All intergrated circuits have an internal cooldown, to protect from spam.
+		return
+	if(power_draw_per_use && !ignore_power)
+		if(!check_power())
+			power_fail()
+			return
+	next_use = world.time + cooldown_per_use
```

Comment:
More time stuff. You can probably remove this whole proc..

Reply:
This proc is handling cooldown of circuits, Do you want to remove cooldown entirely?